### PR TITLE
boards: frdm_k64f: Solve conflict between UART3 and enet

### PR DESCRIPTION
**Pull request by KimBP**
Arduino compatible boards should have a working UART on pin 0 & 1.
On frdm_k64f this is handled by uart3

Unfortunately pinmuxing of uart3 was destroyed when enabling NETWORKING,
because the uart3 RX/TX pins was reconfigured for 1588 timers.

1588 timers are enabled by the enet child node, ptp which by some reason
is enabled by default.
1588 timers aren't needed in most cases when ethernet is being used,
so this fix ensures ptp is by default disabled. Likewise pinmuxing of
the 1588 timer functionality is now dependant of ptp being active

Signed-off-by: Kim Bøndergaard <kim.boendergaard@escoglobal.com>

### CODE REVIEW

File: boards/arm/frdm_k64f/frdm_k64f.dts
Comment (mmahadevan108) [inline boards/arm/frdm_k64f/frdm_k64f.dts:243]: A comment over here about the pin conflict will be helpful

**Comment by mmahadevan108** [comment]
@hakehuang, could you kindly have a look at this change to disable PTP on FRDM-K64F boards.

File: boards/arm/frdm_k64f/frdm_k64f.dts
Comment (hakehuang) [reply boards/arm/frdm_k64f/frdm_k64f.dts:243]: ptc16/17 is use as tmr event for compare and capture, which is not a must in 1588. so we can disable the pinmux in pinmux.c instead of disable ptp in frdmk64f board.

File: boards/arm/frdm_k64f/pinmux.c
Comment (hakehuang) [inline boards/arm/frdm_k64f/pinmux.c:107]: may we can add a config for ENABLE_PTP_EVENT_PINS to conctrol this pinmux, not use the device node

**Comment by hakehuang** [comment]
> @hakehuang, could you kindly have a look at this change to disable PTP on FRDM-K64F boards.

@mmahadevan108 , this is ok, as @KimBP enable the ptp on gptp sample. as the timer_event capture/compare pin is not a must in gptp, so we may use a config to control the pinmux and enalbe the ptp timer by default.

File: boards/arm/frdm_k64f/pinmux.c
Comment (KimBP) [reply boards/arm/frdm_k64f/pinmux.c:107]: I'll vote against yet a config. My solution does it seamless. I can't see how it becomes better by adding a config. 
Most other pinmux entries in the frdm__k64f board are already based on DT_ macros.
Enabling ptp in the device tree without defining pins doesn't really make sense - or does it (Refer you previous comment)?
Hopefully in the (near) future all pinmuxing is found in the dts making issues like this more transparent.

Comment (dleach02) [reply boards/arm/frdm_k64f/pinmux.c:107]: I tend to agree that using device tree for this is more inline with the intent of device tree.

**Review by dleach02 — CHANGES_REQUESTED** [review]
Add comments in the board dts about the conflict

**Comment by KimBP** [comment]
I've added a comment about conflicting pins like requested

**Review by mmahadevan108 — APPROVED** [review]

**Review by hakehuang — APPROVED** [review]

**Comment by KimBP** [comment]
@dleach02 May I ask you to follow up on this PR

**Comment by KimBP** [comment]
@hakehuang @mmahadevan108 Can I do any more on this PR?.I can't find any way to change the 'request change' from @dleach02 to 'resolved'. How do we proceed?

**Comment by hakehuang** [comment]
@KimBP need @dleach02 confirm you updates, please wait for approval.

**Review by dleach02 — APPROVED** [review]

**Comment by lammens** [comment]
wrong site but how can I decode from the AIS.h library on a Arduino nano extract the shipname. please help.

**Comment by KimBP** [comment]
Are you referring to my AIS library at <https://github.com/KimBP/AIS> then please ask via git hub issues (but get_shipname() is a good place to start looking)

/KimBP

**Comment by lammens** [comment]
Yes, I´m trying to decode live AIS messages with a Aduino-Nano, and yes I know, the Arduino is not suitable for this but I only need the position and shipname which is ok for me. But I get no shipname using "   String shipnaam; 
 shipnaam = is_msg.get_shipname(); "
I have very little knowlidge of C or C++ and I´m lost in the ais.h and ais.cpp files.
I get the proper  MMSI, SOG, LONG, LAT .
Plese be so kind to help me.
Thanks and regards. Hans L.

**Pull request merged**
